scrapypuppeteer/middleware.py

```diff
@@ -227,25 +266,33 @@ def process_response(self,
         if not isinstance(response, PuppeteerResponse):  # We only work with PuppeteerResponses
             return response
 
-        if response.puppeteer_request.close_page:  # No need in solving captcha right before closing the page
+        if request.meta.get('dont_recaptcha', False):  # Skip such responses
             return response
 
         if request.meta.pop('_captcha_submission', False):  # Submitted captcha
-            return self._gen_response(response)
+            return self.__gen_response(response)
 
         if isinstance(response.puppeteer_request.action, RecaptchaSolver):
             # RECaptchaSolver was called by recaptcha middleware
             return self._submit_recaptcha(request, response, spider)
 
+        if isinstance(response.puppeteer_request.action, (Screenshot, Scroll, CustomJsAction)):
+            # No recaptcha after these actions
+
+            # TODO: I guess there is almost no recaptcha after Click actions
+            # We can add '_captcha_solving' meta key to determine if RecaptchaSolver is ours
+            # So, after this user is able to call RecaptchaSolver on its own
+            return response
```

Comment:
I disagree with scrolls. I can think of a blog page where you can endlessly scroll for new posts triggering new ajax requests, and if you do so long/fast enough, you may receive captcha.

Not sure about custom actions though. On the one hand, such actions may implement any sorts of browser interactions that provoke captchas. On the other hand, we may leave their handling on users, since they form the responses, so they can check for captchas explicitly. @Yatskov Alex, what do you think?